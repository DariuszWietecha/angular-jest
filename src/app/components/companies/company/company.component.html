<div class="card-body">
  <h4 class="card-title">{{ company.name }}</h4>

  <dt>Name:</dt>
  <dd>{{ company.name }}</dd>

  <dt>Logo:</dt>
  <dd>{{ company.logoUrl }}</dd>

  <dt>E-mail:</dt>
  <dd>{{ company.email }}</dd>

  <dt>Categories:</dt>
  <dd>
    <ul>
      <li *ngFor="let category of company.categories">{{category.name}}</li>
    </ul>
  </dd>

  <a [routerLink]="['/companies/edit', company.id]" class="btn btn-primary"  *ngIf="showEditButton === true">Edit</a>
</div>