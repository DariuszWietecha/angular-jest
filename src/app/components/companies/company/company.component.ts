import { Component, Input, HostBinding } from '@angular/core';
import { ICompany } from '../../../services/companies.service';

@Component({
  selector: 'app-company',
  templateUrl: './company.component.html',
  styleUrls: ['./company.component.scss']
})

export class CompanyComponent {
  @HostBinding('class') className = 'card col-4';
  @Input() company: ICompany;
  @Input() showEditButton = true;
  constructor() { }
}
