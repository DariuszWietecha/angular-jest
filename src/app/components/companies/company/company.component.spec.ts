import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { CompanyComponent } from './company.component';

describe('CompanyComponent', () => {
  let component: CompanyComponent;
  let fixture: ComponentFixture<CompanyComponent>;
  const comapaniesListMock = require('../../../../../test/data/companies/companies-get.json');

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [CompanyComponent]
    })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CompanyComponent);
    component = fixture.componentInstance;
    component.company = comapaniesListMock[0];
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapShot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
