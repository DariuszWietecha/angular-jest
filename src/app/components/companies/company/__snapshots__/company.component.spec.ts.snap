// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyComponent snapShot 1`] = `
<app-company
  className={[Function String]}
  company={[Function Object]}
  showEditButton={[Function Boolean]}
>
  <div
    class="card-body"
  >
    <h4
      class="card-title"
    >
      Great Company
    </h4>
    <dt>
      Name:
    </dt>
    <dd>
      Great Company
    </dd>
    <dt>
      Logo:
    </dt>
    <dd>
      Great cube
    </dd>
    <dt>
      E-mail:
    </dt>
    <dd>
      mail@greatcompany.com
    </dd>
    <dt>
      Categories:
    </dt>
    <dd>
      <ul>
        
        <li>
          Big Category
        </li>
      </ul>
    </dd>
    
    <a
      class="btn btn-primary"
      href="/companies/edit/f0a6d4b0-6591-11ea-a253-fd7829c2ab13"
      ng-reflect-router-link="/companies/edit,f0a6d4b0-6591-"
    >
      Edit
    </a>
  </div>
</app-company>
`;
